rgw/lua: Fix Lua background thread execution timing and config updates (#61560)
ceph/ceph: doc/radosgw/lua-scripting.rst

To change this default value, use the ``rgw_lua_max_memory_per_state`` configuration parameter. Note that the basic overhead of Lua with its standard libraries is ~32K bytes. To disable the limit, use zero.
By default, the execution of a Lua script is limited to a maximum runtime of 1000 milliseconds. This limit can be changed using the ``rgw_lua_max_runtime_per_state`` configuration parameter. If a Lua script exceeds this runtime, it will be terminated. To disable the runtime limit, use zero.

> .. warning:: Be cautious when modifying the memory limit. If the current memory usage exceeds the newly set limit, the Lua state will be restarted, resulting in the loss of previously stored data.

Reviewer: ```suggestion
.. warning:: Be cautious when modifying the memory limit. If the current memory usage exceeds the newly set limit, all previously stored data in the background state will be lost.
```

> .. warning:: Disabling the runtime limit may result in unbounded script execution, which can lead to excessive resource consumption and potentially impact Ceph cluster stability or availability.

Reviewer: ```suggestion
.. warning:: Disabling the runtime limit may result in unbounded script execution, which can lead to excessive resource consumption and potentially impact the RADOS gateway's availability.
```
